internal/jimmhttp/http.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024 Canonical.
+
+package jimmhttp
+
+import (
+	"context"
+	"net/http"
+)
+
+type HTTPHandler struct {
+	HTTPProxier HTTPProxier
+}
+
+func (h *HTTPHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	if h.HTTPProxier == nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	h.HTTPProxier.ServeHTTP(ctx, w, req)
```

Comment:
We have a servehttp here with the wrong signature and embed it in the handler, why?